src/WCSimPrimaryGeneratorAction.cc

```diff
@@ -414,6 +419,70 @@ void WCSimPrimaryGeneratorAction::GeneratePrimaries(G4Event* anEvent)
       particleGun->GeneratePrimaryVertex(anEvent);
     }//old muline format
   }//useMuLineEvt
+  
+  else if(useAmBeEvt){ // Diego Costas (diego.costas.rodriguez@usc.es) 2023
+    // Create the energy distribution object
+    G4SPSEneDistribution* nEnergyDist = new G4SPSEneDistribution();
+    G4SPSRandomGenerator* rGen = new G4SPSRandomGenerator();
+
+    // Set the neutron energy distribution from a file
+    nEnergyDist->SetEnergyDisType("Arb");
+    nEnergyDist->ArbEnergyHistoFile("../data/resampled_nSpectrum.txt");
+    nEnergyDist->SetBiasRndm(rGen);
+    nEnergyDist->ArbInterpolate("Lin");
+
+    // Obtain the definition of the neutron and sample an energy from the distribution
+    G4ParticleTable* particleTable = G4ParticleTable::GetParticleTable();
+    G4ParticleDefinition* neutronDefinition = particleTable->FindParticle("neutron");
+    //G4ParticleDefinition* neutronDefinition = G4Neutron::Definition();
+    G4double nEnergy = nEnergyDist->GenerateOne(neutronDefinition);
+
+    // Define PDG IDs, energies and desired particle names
+    G4int pdgids[] = {2112, 22};
+    G4double energys[] = {nEnergy, 0.0};
+    G4String particleNames[] = {"neutron", "gamma"};
+
+    // Definir las probabilidades correspondientes a cada energía de partícula gamma
+    std::vector<double> probabilities = {0.26, 0.65, 0.08};
+    std::vector<double> energies = {0.0, 4.4, 7.7};
+
+    // Generar un número aleatorio ponderado para seleccionar la energía de la partícula gamma
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::discrete_distribution<int> dist(probabilities.begin(), probabilities.end());
+    int index = dist(gen);
+    G4double gammaEnergy = energies[index];
+
+    for (G4int i=0; i<2; i++){
+      // Create a vertex for the particle following the index of the loop
+      G4int pdgid = pdgids[i];
+      G4ThreeVector vtx = G4ThreeVector(0., 0., 0.);
+      G4double time = 0.;
+      G4double ekin = (i == 0) ? nEnergy : gammaEnergy;
+
+      // Configure particle's properties in particleGun
+      if (pdgid == 22) {
+        particleGun->SetParticleDefinition(particleTable->FindParticle("gamma"));
```

Comment:
It'd be nice to cache the `photonDefintion` (like you've cached `neutronDefinition`)
(I'm being pedantic here since string comparison is slow)